### ISSUE TITLE
[Form] Query builder is run for each entity within each instance of a form within a collection

### ISSUE THREAD

**Issue by kriswillis** [symfony/symfony#8658]
When building a form that has a collection field that contains a form with entity fields using the `query_builder` option, the query is run for each instance in the collection.

Example:
```php
// Main form
public function buildForm(FormBuilderInterface $builder, array $options)
{
    $builder
    ->add('participants', 'collection', array(
        'type'         => new ParticipantType(),
        'allow_add'    => true,
        'allow_delete' => true,
        'by_reference' => false,
    ));
}

// Form used in collection (ParticipantType)
public function buildForm(FormBuilderInterface $builder, array $options)
{
    $builder
    ->add('user', 'entity', array(
        'class' => 'MyUserBundle:User',
        'query_builder' => function(EntityRepository $er) {
            return $er->createQueryBuilder('u')
                ->orderBy('u.surname', 'ASC');
        },
    ))
    ->add('relationship', 'entity', array(
        'class' => 'MyOtherBundle:Relationship',
        'query_builder' => function(EntityRepository $er) {
            return $er->createQueryBuilder('r')
                ->orderBy('r.name', 'ASC');
        },
    ));
}
```

Now, if I generate this form with an entity that has 30 `participants`, the query builder is run for both the `user` and `relationship` entity fields 30 times, resulting in 60 queries. Surely each query only needs to be run once?

**Comment by lokhman** [comment]
Any updates on this? The issue still exists, and I wonder why it was closed.

**Comment by xabbuh** [comment]
@lokhman Which Symfony version do you use? Do the changes made in #16747 improve your situation?

**Comment by lokhman** [comment]
@xabbuh It was 2.7.7. Updated to 2.7.9, and now it's fixed. Thank you for reference. The issue was resolved a week ago and released with [**v2.7.9 #17367**](https://github.com/symfony/symfony/issues/17367).

**Comment by xabbuh** [comment]
@lokhman Thanks for the confirmation.
